packages/kyt-core/src/config/webpackHotDevClient.js

```diff
@@ -0,0 +1,273 @@
+      protocol: window.location.protocol,
+      hostname: window.location.hostname,
+      port: sockJSPort,
+      pathname: launchEditorEndpoint,
+      search: `?fileName=${window.encodeURIComponent(
+        errorLocation.fileName
+      )}&lineNumber=${window.encodeURIComponent(
+        errorLocation.lineNumber || 1
+      )}&colNumber=${window.encodeURIComponent(errorLocation.colNumber || 1)}`,
+    }),
+    { mode: 'no-cors' }
+  );
+});
+
+// We need to keep track of if there has been a runtime error.
+// Essentially, we cannot guarantee application state was not corrupted by the
+// runtime error. To prevent confusing behavior, we forcibly reload the entire
+// application. This is handled below when we are notified of a compile (code
+// change).
+// See https://github.com/facebookincubator/create-react-app/issues/3096
+let hadRuntimeError = false;
+ErrorOverlay.startReportingRuntimeErrors({
+  onError() {
+    hadRuntimeError = true;
+  },
+  filename: process.env.REACT_BUNDLE_PATH || '/static/js/bundle.js',
+});
+
+if (module.hot && typeof module.hot.dispose === 'function') {
+  module.hot.dispose(() => {
+    // $TODO: why do we need this?
+    ErrorOverlay.stopReportingRuntimeErrors();
+  });
+}
+
+// Connect to WebpackDevServer via a socket.
+const connection = new SockJS(
+  url.format({
+    protocol: window.location.protocol,
+    hostname: window.location.hostname,
+    port: sockJSPort,
+    // Hardcoded in WebpackDevServer
+    pathname: '/sockjs-node',
+  })
+);
+
+// Unlike WebpackDevServer client, we won't try to reconnect
+// to avoid spamming the console. Disconnect usually happens
+// when developer stops the server.
+connection.onclose = () => {
+  if (typeof console !== 'undefined' && typeof console.info === 'function') {
+    console.info('The development server has disconnected.\nRefresh the page if necessary.');
+  }
+};
+
+// Remember some state related to hot module replacement.
+let isFirstCompilation = true;
+let mostRecentCompilationHash = null;
+let hasCompileErrors = false;
+
+function clearOutdatedErrors() {
+  // Clean up outdated compile errors, if any.
+  if (typeof console !== 'undefined' && typeof console.clear === 'function') {
+    if (hasCompileErrors) {
+      console.clear();
+    }
+  }
+}
+
+// Successful compilation.
+function handleSuccess() {
+  clearOutdatedErrors();
+
+  const isHotUpdate = !isFirstCompilation;
+  isFirstCompilation = false;
+  hasCompileErrors = false;
+
+  // Attempt to apply hot updates or reload.
+  if (isHotUpdate) {
+    tryApplyUpdates(function onHotUpdateSuccess() {
+      // Only dismiss it when we're sure it's a hot update.
+      // Otherwise it would flicker right before the reload.
+      ErrorOverlay.dismissBuildError();
+    });
+  }
+}
+
+// Compilation with warnings (e.g. ESLint).
+function handleWarnings(warnings) {
+  clearOutdatedErrors();
+
+  const isHotUpdate = !isFirstCompilation;
+  isFirstCompilation = false;
+  hasCompileErrors = false;
```

Comment:
Why do we set this to `false` here? Will we simply not get to this line of code if there are compile errors?

Reply:
the flag gets reset on hot compilations